cmd/telegraf/telegraf.go

```diff
@@ -152,6 +153,9 @@ func runAgent(ctx context.Context,
 ) error {
 	// Setup default logging. This may need to change after reading the config
 	// file, but we can configure it to use our logger implementation now.
+	if !initLoggingSet {
+		logger.SetupLogging(logger.LogConfig{})
```

Comment:
Removing this would wipe the `evenlog` setting created in the service section. 
Other option is to have a global  variable for initial default config and modify it just for the service mode. The var will than be used here.

Reply:
I don't follow, wouldn't this only be called if the event log couldn't be initialized?  In that case this has already be called in main.

Reply:
[In this part](https://github.com/bonitoo-io/telegraf/blob/vh-startup-error-reporting/cmd/telegraf/telegraf.go#L450) ,
``` go
           } else {
			winlogger, err := s.Logger(nil)
			if err == nil {
				//When in service mode, register eventlog target andd setup default logging to eventlog
				logger.RegisterEventLogger(winlogger)
				logger.SetupLogging(logger.LogConfig{LogTarget: logger.LogTargetEventlog})
				initLoggingSet = true
			}
			err = s.Run()

			if err != nil {
				log.Println("E! " + err.Error())
			}
		}
```
I'm setting default config option to `eventlog` **only** when telegraf runs on windows as a service and it runs before [`runAgent`](https://github.com/bonitoo-io/telegraf/blob/vh-startup-error-reporting/cmd/telegraf/telegraf.go#L157),
``` go
func runAgent(ctx context.Context,
	inputFilters []string,
	outputFilters []string,
) error {
	// Setup default logging. This may need to change after reading the config
	// file, but we can configure it to use our logger implementation now.
	if !initLoggingSet {
		logger.SetupLogging(logger.LogConfig{})
	}
	log.Printf("I! Starting Telegraf %s", version)
``` 
which runs always.
If there is no check for pre-set default logging, this part will re-set previously set eventlog to stderr

Reply:
I'm not saying remove the check, I think we can remove all of this change.  If we remove it then Windows will be fine and wouldn't non-Windows also be okay because we are now calling SetupLogging earlier in main?

Reply:
I overlooked setting up default logging in main. So, sure, this part in `runAgent` so useless.